File: frontend/src/types/recipe.ts
export type RecipeCategory =
  | "Breakfast"
  | "Lunch"
  | "Dinner"
  | "Dessert"
  | "Snack";

export type RecipeSummary = {
  id: string;
  title: string;
  slug: string;
  description: string;
  category: RecipeCategory;
  prepTimeMinutes: number;
  servings: number;
  imageUrl: string;
  isPublished: boolean;
};

export type RecipeDetail = RecipeSummary & {
  ingredients: string[];
  steps: string[];
};

File: backend/RecipeApp.Api/Domain/Recipe.cs
namespace RecipeApp.Api.Domain;

public sealed record Recipe
{
    public required Guid Id { get; init; }
    public required string Title { get; init; }
    public required string Slug { get; init; }
    public required string Description { get; init; }
    public required string Category { get; init; }
    public required int PrepTimeMinutes { get; init; }
    public required int Servings { get; init; }
    public required string ImageUrl { get; init; }
    public required bool IsPublished { get; init; }
    public required IReadOnlyList<string> Ingredients { get; init; }
    public required IReadOnlyList<string> Steps { get; init; }
}
